core/src/main/java/org/eclipse/hono/config/ProtocolAdapterProperties.java

```diff
@@ -13,17 +13,27 @@
 
 package org.eclipse.hono.config;
 
+import java.time.Duration;
+import java.util.Objects;
+
 /**
  * Common configuration properties for protocol adapters of Hono.
  *
  */
 public class ProtocolAdapterProperties extends ServiceConfigProperties {
 
+    /**
+     * The default duration after which a tenant times out when no message has been sent for it. The default value
+     * disables automatic tenant timeout.
+     */
+    public static final Duration DEFAULT_IDLE_TENANT_TIMEOUT = Duration.ZERO;
+
     private boolean authenticationRequired = true;
     private boolean jmsVendorPropsEnabled = false;
     private boolean defaultsEnabled = true;
     private long eventLoopBlockedCheckTimeout = 5000L;
     private int maxConnections = 0;
+    private Duration idleTenantTimeout = DEFAULT_IDLE_TENANT_TIMEOUT;
```

Comment:
`idleTenantTimeout` sounds a bit odd to me, how about using `tenantIdleTimeout` instead?